src/CurrentSelection/CurrentSelection.tsx

```diff
@@ -76,7 +105,7 @@ class CurrentSelection extends React.Component<
         </section>
         <button
           className="create-team-button"
-          onClick={(): void => this.postNewTeam()}
+          onClick={(): void | Promise<void> => this.postNewTeam()}
```

Comment:
you don't need to declare this return type if you already define it in `postNewTeam`.